### PR TITLE
[GEOS-10287] loadJSON function is added to OSEO-STAC

### PR DESCRIPTION
**Pull request by ydeliorman**
[GEOS-10287](https://osgeo-org.atlassian.net/browse/GEOS-10287)

loadJSON functionality is added to OSEO-STAC module which enables to load a JSON file in geoserver data directory.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/oseo/oseo-stac/src/test/java/org/geoserver/ogcapi/stac/LoadJSONTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Test
+    public void testReadAndEvalJSON() throws Exception {
+        Document dom = getAsJSoup("ogc/stac/collections/SENTINEL2?f=html");
+
+        assertEquals("attribute1 => 1 attribute2 => 2", dom.select("h2").text());
+    }
+
+    @Test
+    public void testReadFileOutsideOfDataDir() throws Exception {
```
Comment (taba90) [inline src/community/oseo/oseo-stac/src/test/java/org/geoserver/ogcapi/stac/LoadJSONTest.java:40]: Seems to me that this method doesn't test what it claims to test. Could you clarify, fix it?

**Review by taba90 — COMMENTED** [review]
Overall looks good to me.
I've only a point regarding the test for checking that file loading from outside the data dir is not permitted. See below.

**Comment by ydeliorman** [comment]
review is done

**Review by taba90 — APPROVED** [review]
The pr looks fine. Tests have been added also to check that file outside the data dir are not loaded. Run the build locally and tests are all passing.

**Pull request merged**